typescript/test/unit/proto.ts

```diff
@@ -251,6 +252,25 @@ describe('src/schema/proto.ts', () => {
     });
   });
 
+  describe('should return a string set to camelCase', () => {
+    it('should return this to camelCase', () => {
+      assert.deepStrictEqual(
+        convertFieldToCamelCase('name.name2.name3'),
+        'name.name2.name3'
+      );
+      assert.deepStrictEqual(convertFieldToCamelCase(''), '');
+      assert.deepStrictEqual(convertFieldToCamelCase('parent_id'), 'parentId');
+      assert.deepStrictEqual(
+        convertFieldToCamelCase('app_profile_id'),
+        'appProfileId'
+      );
+      assert.deepStrictEqual(
+        convertFieldToCamelCase('name.parent_id.another_parent_id'),
+        'name.parentId.anotherParentId'
+      );
```

Comment:
Can we add a unit test on a field name with non-alphanumeric letter like `name?_(1)`. I expect the result is same `name?_(1)`.

Reply:
I've added it, but based on how the `toCamelCase` function is defined, `name?_(1)` actually would get converted to `name_1`, because it splits on non-alphanumeric characters. This should be okay in our case.

in `util.ts`:

```
String.prototype.words = function (this: string): string[] {
  // split on spaces, non-alphanumeric, or capital letters
  return this.split(/(?=[A-Z])|[\s\W_]+/)
    .filter(w => w.length > 0)
    .map(w => w.toLowerCase());
};

String.prototype.toCamelCase = function (this: string): string {
  const words = this.words();
  if (words.length === 0) {
    return this;
  }
  const result = [words[0]];
  result.push(
    ...words.slice(1).map(w => {
      if (isDigit(w)) {
        return '_' + w;
      }
      return w.capitalize();
    })
  );
  return result.join('');
};
```

Reply:
`name?_(1)` -> `name_1` is not correct. But let's move the test case for now. I hope that no client use such wired name. If it comes later, I will take care of it . Thank you.

Reply:
Got it, sounds good. Removed.